### DISCUSSION TITLE
Example in Looping Automation using JS

### DISCUSSION THREAD

**Discussion by ducnhoxlatui** [Budibase/budibase#10986] [Help] [answered]
Hi guys, 
can you guys provide me some example about using JS in looping automation. Cause i try to use js function but it cannot work:

return  $("loop.currentItem.Name") 

Thanks for your supporting <3

**Comment by ducnhoxlatui** [comment]
this is my full code js:
let text = $("loop.currentItem.Concern")
let title = "Technology, Marketing, Sale"

let welcometo = $("steps.5.rows.[0].Welcome")
let end = $("steps.5.rows.[0].Content")
let head = $("steps.5.rows.[0].Head")
let yourContent= "<p>"

if (title.includes(text))
  {
    if (text.includes("Technology"))
      {
       yourContent = yourContent + $("steps.2.rows.[0].Description") + "<br>"
      }
    if (text.includes("Marketing"))
      {
       yourContent = yourContent + $("steps.3.rows.[0].Description") + "<br>"
      }
    if (text.includes("Sale"))
      {
       yourContent = yourContent + $("steps.4.rows.[0].Description") + "<br>"
      }
  }
return welcometo +" "+ $("loop.currentItem.Name") + head + yourContent +"</p>"+ end


Beside that, i only use step 1 to loop, so i use step 2,3,4,5 is ok, right?

**Reply by andz-bb** [reply]
hey @ducnhoxlatui, I was able to get it working using this code:

```js
const text = $("loop.currentItem.Concern");
const welcometo = $("steps.5.rows.[0].Welcome") + " " + $("loop.currentItem.Name");
const end = $("steps.5.rows.[0].Content");
const head = $("steps.5.rows.[0].Head");

const concerns = text.split(',').map(concern => concern.trim());

const descriptions = {
    "Technology": $("steps.2.rows.[0].Description"),
    "Marketing": $("steps.3.rows.[0].Description"),
    "Sale": $("steps.4.rows.[0].Description"),
};

const yourContent = concerns.map(concern => descriptions[concern]).join("</br>");

return `${welcometo}${head}${yourContent}${end}
```

Hope this helps - if you're still having problems let me know and I can do some more investigation

**Reply by ducnhoxlatui** [reply]
![image](https://github.com/Budibase/budibase/assets/71976420/2dbbdd5b-fc39-438f-b2c4-3cc83c2ed018)

Its still caused error :((( @andz-bb

**Reply by andz-bb** [reply]
@ducnhoxlatui Could you send me your latest app export please? I need to check that none of the other steps in the automation have changed. I also had made a copy of your Google Sheets document for my own testing. Have you changed any of the columns on that since last week? Thanks

**Reply by andz-bb** [reply, marked as answer]
@ducnhoxlatui 

the last character ` got cut off when I pasted the new script to you. I'm very sorry about that! I've tested this now on your new app export and it is working as expected. Here's the full script:

```js
const text = $("loop.currentItem.Concern");
const welcometo = $("steps.5.rows.[0].Welcome") + " " + $("loop.currentItem.Name");
const end = $("steps.5.rows.[0].Content");
const head = $("steps.5.rows.[0].Head");

const concerns = text.split(',').map(concern => concern.trim());

const descriptions = {
    "Technology": $("steps.2.rows.[0].Description"),
    "Marketing": $("steps.3.rows.[0].Description"),
    "Sale": $("steps.4.rows.[0].Description"),
};

const yourContent = concerns.map(concern => descriptions[concern]).join("</br>");

return `${welcometo}${head}${yourContent}${end}`;
```

**Reply by ducnhoxlatui** [reply]
Wow, its worked, thank @andz-bb !!!
